tests/dummy/app/templates/index-example.hbs

```diff
@@ -0,0 +1,53 @@
+<section>
+  <h1>Create a new user</h1>
+  <form
+    id="userForm"
+    {{verify-form-validity submit=this.handleFormSubmit}}
+    {{on "validated" this.assignValidationMessages}}
+  >
+    <div class="full-width text-right"><sup class="required">*</sup> Required</div>
+
+    <label for="userForm-name">Name<sup class="required">*</sup></label>
+    <input id="userForm-name" name="name" required {{validity}}>
+    <span class="validation-message">{{this.validationMessages.name}}</span>
+
+    <label for="userForm-phone">Phone</label>
+    <input id="userForm-phone" name="phone" type="tel" {{validity}}>
+    <span class="validation-message">{{this.validationMessages.phone}}</span>
+
+    <label for="userForm-email">Email<sup class="required">*</sup></label>
+    <input id="userForm-email" name="email" type="email" required {{validity}}>
+    <span class="validation-message">{{this.validationMessages.email}}</span>
+
+    <label for="userForm-password">Password<sup class="required">*</sup></label>
+    <input
+      id="userForm-password"
+      name="password"
+      type="password"
+      required
+      {{validity (validate-password)}}
+    >
+    <span class="validation-message">{{this.validationMessages.password}}</span>
+
+    <label for="userForm-confirm">Confirm password<sup class="required">*</sup></label>
+    <input
+      id="userForm-confirm"
+      name="confirm"
+      type="password"
+      required
+      {{validity (validate-confirmation "userForm-password")}}
```

Comment:
For demo purposes, assuming the password field set some property on the backing controller on input, this could leverage `validateTracked` to error when the first field changes.

Reply:
That would only be a need if something outside this component modified the tracked property. Also this example has no backing data as it relies on the native FormData when the form submits.

Are you suggesting we use a different example to demonstrate the weirdness of Data Up/Down dynamics of tracked properties?

Reply:
_this might be moot based on the below thread, but, just in case..._

Yeah, this would suggest inventing a tracked property (or two) that the password field syncs to on input. As it is now, if the user were to set both password fields to the same, then change the first password field (thereby causing both to no longer equal), the second doesn't error until the next input/change/submit event.

It's a bit of a contrived example, but it demonstrates how to handle keeping two things (which could be two components coordinating through DDAU) in lockstep with each other.

Reply:
Ahh I see what you mean. Yes that could be managed via tracked properties and the `validateTracked` but it could also be managed vanilla by making the password also have a validation to check the reverse case. Updating example.

Reply:
I played with this myself and I think you are right. The password/confirmation is a prime example of needing a tracked property and to validate on it being dirtied. I hope my helper `chage-tracker` illustrates this unique but necessary situation.

The only alternative I could imagine is to provide a `validate` modifier which could dispatch the `validate` DOM event but at that point I think the addon expands too far from the Ember expectations and ecosystem. I think that the business logic of related validators is best kept within the system it is being used. VanillaJS can use the `validate` function while Ember can use tracked properties through the `validateTracked` param.

I think I'm ok with the API as it is currently in this PR. We can refine this later.

I also think this situation and a few other threads here illustrate the power of the underlying design of DOM events and custom validity that it is flexible enough to have more then one way to skin the cat. 😉